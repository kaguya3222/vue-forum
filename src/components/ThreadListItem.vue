<template>
  <div class="thread">
    <div>
      <p>
        <router-link
          :to="{ name: 'ThreadShow', params: { threadId: thread['.key'] } }"
          >{{ thread.title }}</router-link
        >
      </p>
      <p class="text-faded text-xsmall">
        By <a href="#">{{ user.name }}</a
        >,
        <app-date :unixDate="thread.publishedAt" />
      </p>
    </div>

    <div class="activity">
      <p class="replies-count">{{ repliesCount }} replies</p>
    </div>
  </div>
</template>

<script>
import { countObjectProperties } from "../helpers";

export default {
  props: {
    thread: {
      required: true,
      type: Object
    }
  },
  name: "ThreadListItem.vue",
  computed: {
    repliesCount() {
      return countObjectProperties(this.thread.posts) - 1;
    },
    user() {
      return this.$store.state.users[this.thread.userId];
    }
  }
};
</script>
