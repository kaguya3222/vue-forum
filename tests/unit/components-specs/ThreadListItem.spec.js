import { createLocalVue, shallowMount } from "@vue/test-utils";
import ThreadListItem from "../../../src/components/ThreadListItem";
import Vuex from "vuex";
import usersGetters from "@/store/modules/users/getters";
import threadsGetters from "@/store/modules/threads/getters";
import mockedRootStore from "../mocks/mockedRootStore";

const localVue = createLocalVue();
localVue.use(Vuex);

const rootState = mockedRootStore.state;

describe("ThreadListItem", () => {
  const store = new Vuex.Store({
    modules: {
      users: {
        namespaced: true,
        state: rootState.users,
        getters: { ...usersGetters }
      },
      threads: {
        namespaced: true,
        state: { ...rootState.threads },
        getters: { ...threadsGetters }
      }
    }
  });
  const threads = mockedRootStore.state.threads.items;
  const wrapper = shallowMount(ThreadListItem, {
    propsData: {
      thread: Object.values(threads)[0]
    },
    stubs: {
      "app-date": true,
      "router-link": true
    },
    localVue,
    store
  });
  test("Shows thread data in threads list", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
