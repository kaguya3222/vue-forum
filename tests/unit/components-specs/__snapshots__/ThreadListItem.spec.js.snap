// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThreadListItem Shows thread data in threads list 1`] = `
<div class="thread">
  <div>
    <p>
      <router-link-stub to="[object Object]">What kind of pet do you have?</router-link-stub>
    </p>
    <p class="text-faded text-xsmall">
      By <a href="#">Rolf</a>,
      <app-date-stub unixdate="1504035908"></app-date-stub>
    </p>
  </div>
  <div class="activity">
    <p class="replies-count">3 replies</p>
  </div>
</div>
`;
